### DISCUSSION TITLE
[New Feature] Option to Lock a Note to “Lock” the Graph Analysis Panel

### DISCUSSION THREAD

**Discussion by FelipeRearden** [SkepticMystic/graph-analysis#33] [Ideas]
Hello @SkepticMystic  !!!!

I have an idea based on my current workflow to share with you :) 

The idea/FR is simple to explain: I would like to have an option to lock/freeze a note in a way that if I changed my active note, the graph panel WILL NOT change/update to the active note:
- A checkbox on the right side of `Refresh Index` to toggle this feature;
- A command palette command to be able to do this with a hotkey command.

## My Workflow 

I’m gonna use the Joe Biden example :)

![0](https://user-images.githubusercontent.com/85180696/141286927-eb7cae76-e8a8-4653-8773-aab0e91d43c3.png)

I want to write a text about Joe Biden. What I do is:

A. I would open my [[Joe Biden]] note (active Note)

B. Start the Research in my Vault: Open Graph Analysis Panel in the Co-Citations Type to find all notes that are related to the [[Joe Biden]] note;

C. Open in a new pane (vertically or horizontally - see #31 ) these related notes found in the Graph Panel [[China]] [[Donald Trump]] [[Barack Obama]] 

D. Navigate (read, highlight) in these notes to understand what I found about Joe Biden that is important in this notes

E. Write about Joe Biden in the [[Joe Biden]] Note 

Every time that I change my active note from [[Joe Biden]] to [[China]] [[Donald Trump]] or [[Barack Obama]] I lost the graph information about [[Joe Biden]] 

My idea is to have a way to lock the graph panel in the [[Joe Biden]] note in a way that I can switch to [[China]] and keep the Joe Biden view :)
- This way I can open [[China]] [[Donald Trump]] [[Barack Obama]] in a new pane horizontally or vertically. 

In my specific case I set a specific Workspace using the Workspace core plug-in named “MOC” with a preset configuration for every time that I need to create a note like this:

![1](https://user-images.githubusercontent.com/85180696/141286978-38cb2a98-f24e-4805-ae6f-f2f560fa4330.png)

Then I open the [[Joe Biden]] note + lock [[Joe Biden]] and start to open the notes found in the graph view into to a screen like this:

![2](https://user-images.githubusercontent.com/85180696/141287004-f1c2995c-d9f3-4eee-820c-e258379c8a27.png)

If I need to expand my research to [[Donald Trump]] I can hot the command key to unlock the graph panel and view the relations to the [[Donald Trump]] note :)

—-

Sorry for the long text :) thanks for reading this!

I wait for your comments to create (or not create) an FR about this discussion!

Have a great day!

**Comment by SkepticMystic** [comment]
@FelipeRearden thank you for the FR, we both really like this idea 😄 
And so, it has been implemented in 0.10.0! Please let us know if it is working as expected for you.
There is now a button to toggle `Frozen` mode, which will enable the behaviour you describe:
![image](https://user-images.githubusercontent.com/70717676/141644408-51ca9c87-18a3-4998-a7db-34b903dd5a57.png)

**Reply by FelipeRearden** [reply]
AMAZING !!!!! Thank very much @SkepticMystic and @HEmile !!!!!!!!!!!!!

The icons for `frozen` and `unfrozen` are spectacular :)

I'm testing and it's working perfect on version `0.10.0` in Obsidian `0.12.19`
- I have on tiny comment about switching between types that I will explain in #35 :)

Thank you very much for making this happen. I'm so happy :)
